Helix crashes during editing of ERB files

### Summary

During working on Ruby on Rails project with ERB files, Helix freezes and then exits and indefinitely prints the following (I could not stop that, only closing the terminal helps.)

![image](https://user-images.githubusercontent.com/10872792/185874813-778d40f1-5c4c-4e5f-ad66-b26d7cdc2859.png)

Note to Helix log: I don't have any files named `text.html.erb` in the project.

### Reproduction Steps

I opened a required file with `space` + `f` which was created by `touch`. 

Immediately pressed `i` to get into inserted mode. I wrote the following code:

```
<%= turbo_stream.replace 'scenarios' %>
<% end %>
```
Still being in the insert mode, I used `ctrl` + `left arrow` to get on the begging of the line, then pressed `up arrow` and then `tab` and added a character `<`. After `<` I wanted to write '%' and here the crash happens. So this is where it got stuck and crashed:

```
<%= turbo_stream.replace 'scenarios' %>
	<
<% end %>
```


### Helix log

2022-08-18T12:22:02.566 helix_lsp::transport [ERROR] err <- "Solargraph is listening on stdio PID=11806\n"
2022-08-18T12:27:33.300 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2022-08-19T10:16:55.455 helix_lsp::transport [ERROR] err <- "Solargraph is listening on stdio PID=202627\n"
2022-08-19T11:58:43.747 helix_term::commands [ERROR] Shell error: touch: cannot touch 'app/views/scenarios/index.html.erb': No such file or directory

2022-08-19T11:59:26.933 helix_term::commands [ERROR] Shell error: touch: cannot touch 'app/views/scenarios/index.html.erb': No such file or directory

2022-08-19T12:00:33.030 helix_view::editor [ERROR] Failed to initialize the LSP for `text.html.erb` { LSP not defined }
2022-08-19T12:01:55.370 helix_view::editor [ERROR] Failed to initialize the LSP for `text.html.erb` { LSP not defined }
2022-08-22T10:15:49.998 helix_view::editor [ERROR] Failed to initialize the LSP for `text.html.erb` { LSP not defined }
2022-08-22T10:16:06.095 helix_view::editor [ERROR] Failed to initialize the LSP for `text.html.erb` { LSP not defined }
2022-08-22T10:16:26.594 helix_lsp::transport [ERROR] err <- "Solargraph is listening on stdio PID=33284\n"
2022-08-22T10:17:33.231 helix_view::editor [ERROR] Failed to initialize the LSP for `text.html.erb` { LSP not defined }
2022-08-22T10:19:52.862 helix_view::editor [ERROR] Failed to initialize the LSP for `text.html.erb` { LSP not defined }
2022-08-22T10:20:49.929 helix_view::editor [ERROR] Failed to initialize the LSP for `text.html.erb` { LSP not defined }
2022-08-22T10:24:25.646 helix_view::editor [ERROR] Failed to initialize the LSP for `text.html.erb` { LSP not defined }
2022-08-22T10:38:50.058 helix_view::editor [ERROR] Failed to initialize the LSP for `text.html.erb` { LSP not defined }

### Platform

Linux (Ubuntu 20.04.1 LTS)

### Terminal Emulator

xterm-256color

### Helix Version

helix 22.05 (c18b5426)

--- a/languages.toml
+++ b/languages.toml
@@ -482,7 +482,7 @@ indent = { tab-width = 2, unit = "  " }
 
 [[grammar]]
 name = "ruby"
-source = { git = "https://github.com/tree-sitter/tree-sitter-ruby", rev = "dfff673b41df7fadcbb609c6338f38da3cdd018e" }
+source = { git = "https://github.com/tree-sitter/tree-sitter-ruby", rev = "4c600a463d97e36a0ca5ac57e11f3ac8c297a0fa" }
 
 [[language]]
 name = "bash"

